src/source_code_style.rs

```diff
@@ -171,17 +171,17 @@ fn detect_quote(contents: &str, locator: &SourceCodeLocator) -> Option<Quote> {
 
 /// Detect the line ending style of the given contents.
 fn detect_line_ending(contents: &str) -> Option<LineEnding> {
-    if let Some(position) = contents.find('\n') {
-        let position = position.saturating_sub(1);
-        return if let Some('\r') = contents.chars().nth(position) {
+    return if let Some(position) = contents.find('\r') {
+        if let Some('\n') = contents.chars().nth(position + 1) {
             Some(LineEnding::CrLf)
         } else {
-            Some(LineEnding::Lf)
-        };
-    } else if contents.find('\r').is_some() {
-        return Some(LineEnding::Cr);
-    }
-    None
+            Some(LineEnding::Cr)
+        }
+    } else if contents.find('\n').is_some() {
+        Some(LineEnding::Lf)
+    } else {
+        None
+    };
```

Comment:
Would it be faster to check for the first `\n`, then check if it's preceded by a `\r`, and only check for a `\r` if we don't find _any_ `\n`?

As-is, we have to traverse the entire string for the `\n` case, to learn that there's no `\r`.